language_service/src/protocol.rs

```diff
@@ -92,6 +92,12 @@ pub struct Hover {
     pub span: Range,
 }
 
+#[derive(Debug, PartialEq)]
+pub struct TextEdit {
```

Comment:
If VS Code is the main consumer, could we just match the struct there and call these 'newText' and 'range' rather than 'contents' and 'span'?

See https://code.visualstudio.com/api/references/vscode-api#TextEdit

Reply:
done, although "newText" had to be "new_text" because of Rust naming conventions